### PR TITLE
Implementing Suspend and Resume for webserver

### PR DESCRIPTION
**Pull request by Karthick-Somasundaresan**

### CODE REVIEW

File: WebServer/WebServerImplementation.cpp [hunk 1]
```diff
@@ -668,35 +685,71 @@ POP_WARNING()
...
-                result = _channelServer.Open(2000);
-            }
-
+            uint32_t result = _channelServer.Configure(service->DataPath(), config);
             return (result);
         }
         PluginHost::IStateControl::state State() const override
         {
-            return (PluginHost::IStateControl::RESUMED);
+            PluginHost::IStateControl::state state;
```
Comment (pwielders) [inline WebServer/WebServerImplementation.cpp:710]: Minor, no need to change but for these short construction it is oke to use the Core::SafeSync<Core::CriticalSection> lock(_adminLock);

File: WebServer/WebServerImplementation.cpp [hunk 2]
```diff
@@ -668,35 +685,71 @@ POP_WARNING()
...
             : _channelServer()
+            , _adminLock()
             , _observers()
+            , _state(PluginHost::IStateControl::UNINITIALIZED)
         {
         }
 
-        ~WebServerImplementation() override = default;
+        ~WebServerImplementation() override {
+            for (auto & observer: _observers){
```
Comment (pwielders) [inline WebServer/WebServerImplementation.cpp:695]: There is no need to do this. Better would be to ASSERT to indicate a potential memory leak and then do run through the observeers, but only Release() the interface(), although that will probably nolonger work :-)

**Review by pwielders — CHANGES_REQUESTED** [review]
There is a race condition in the code that requires some attention.

**Comment by MFransen69** [comment]
We'll pick this up in the future again, lower priority for now

**Pull request closed without merging**